### DISCUSSION TITLE
Flashing Wifi img stream example stuck at 99%

### DISCUSSION THREAD

**Discussion by schplitt** [bitcraze/discussions#721] [Q&A] [answered]
Trying to flash the Wi-Fi Example with the steps found on:
https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#update-crazyflie-and-aideck-firmware

But keeps getting stuck on 99% without any error message.

Doesn't matter which firmware version I use.
AI Deck is on Rev D, so I did not update the GAP8 bootloader, as it is not necessary.

**Comment by knmcguire** [comment]
Hi! I converted your question to discussions as that is the more suitable place, especially since this covers many elements of the ecosystem.

We had some users reporting some similar issues as well, have you look through those discussions? https://github.com/orgs/bitcraze/discussions?discussions_q=is%3Aopen+99%25+aideck+label%3A%22AI+deck%22

And you mentioned that you are flashing the wifi example, but the link you send is pointed to the update crazyflie and aideck-firmware through the client. Did that work?

**Reply by schplitt** [reply]
Hi!
I've looked through those discussions a bit, but nothing seemed to be exactly my problem.
My bad, i meant to link this section: https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#flash-wifi-example

The flashing of the crazyflie firmware worked fine. I have not tried updating the aideck-firmware as my aideck has Rev D and is not necessary as stated in the instructions.
I have no jtag cable on hand right now, but I will try updating the aideck firmware next week.

I will update you with my progress. :)

Furthermore, I also tried this: https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/simple-examples/wifi-streamer/
But got stuck on ```make``` after configuring with ```make menuconfig``` as some .h file was missing.
I will look up which .h file it was.

Thank you for your help. :)

**Reply by schplitt** [reply]
I tried to get my hand on a JTAG programmer, but couldn't get one. Are other programmers compatible with the crazyflie than the ones in the instructions?

I also tried flashing the CNN example, but it gets stuck at 0% and not like the Wi-Fi example on 99%.

And the .h file missing when running ```make``` from 
https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/simple-examples/wifi-streamer/ 
is the "core_cm4.h".
```
 fatal error: core_cm4.h: No such file or directory
  817 | #include "core_cm4.h"             /* Cortex-M4 processor and core peripherals */
```

I'm not really sure what is going on. Maybe I set up the VM wrong? But I never did anything else besides following the instructions.

Help would be appreciated. 

Cheers
Jakob

**Reply by knmcguire** [reply]
You get that error from [ the rebuilding the firmware ](https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/simple-examples/wifi-streamer/#setup-the-wifi-via-the-crazyflie-firmware)  right? It is probably missing some submodules so you will need to do `git submodule init` and `git submodule update`. We should have mentioned that in the documentation more specifically. 

About your issue, you should probably first go through [this tutorial in one piece first](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/), and see if you are able to complete that, before going into the wifi example at all. Could you try that first?

**Reply by schplitt** [reply]
Regarding the tutorial, I already went through it from start to finish, or at least tried to.

Flashing the Crazyflie firmware worked fine without any problems. (Sorry for linking the wrong link in my first message.)
I did not flash the GAP8 firmware as I have a revision d AI Deck, so it was not necessary. (I do not have a JTAG programmer anyway.)
What I got stuck on is the [Flash Wi-Fi example](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#flash-wifi-example):
It flashes the downloaded .bin file until it reaches 99% and just stops. I looked through the posts having the same issue, but nothing seemed to work.

As that didn't work, I went ahead with [setting up the development environment for the GAP8](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#setup-development-environment).
I had no problem with this either.
On a side note, after setting up the environment, the VM storage was full, and the VM wasn't accessible anymore, as is usually the case with VMs with 0 bytes left. You might want to increase the base size of the VM in future versions, as I had to manually increase the size of a new VM as the old one was bricked by having 0 bytes left.

Anyway, after setting up the environment, I went ahead with the [next steps](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#next-steps)

Specifically, with the [Wi-Fi Video Streamer](https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/simple-examples/wifi-streamer/).
There, ```make``` crashed because of the missing .h file.
It makes no difference anyway, as this WI-FI video example is essentially the same as the [other one](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#flash-wifi-example) right?


Afterward, I tried flashing the [classification demo](https://www.bitcraze.io/documentation/repository/aideck-gap8-examples/master/ai-examples/classification-demo/).
The issue here was that the flashing got stuck at 1% and not 99% like in the Wi-Fi Streamer Example.

I'm still trying to get my hands on a JTAG programmer, as this solved the issues for others. But contrary to them, I have a revision D AI deck and not a revision C one or earlier.
I'm really not sure where the problem lies.

**Reply by knmcguire** [reply]
If you can't pass the first tutorial normally, there is no need of trying any other example I'm afraid. so we need to fix that first.

When you flashed the Crazyflie, did you have the AIdeck (as the only deck) attached? Did you see it trying to flash it then? And did you see: `ESP32: I (910) SYS: Initialized,` in the console tab?

**Reply by schplitt** [reply]
The AIdeck is the only deck attached.
The cfclient Bootloader says ```Writing to bcAI:esp deck memory```, so I think the AIdeck is being flashed.
I don't know which LED indicates it, though.
But it eventually disconnects from the Crazyflie, Crazyflie reboots and IDLE is displayed in the bootloader window.

I have tried flashing different versions of the firmware but no different results.
The message in the console tab always says : ```CPX: ESP32: I (434) SYS: Initialized``` and never ```ESP32: I (910) SYS: Initialized```
Additionally, it states:
```
CPX: ESP32: W (434) ROUTER: Cannot route from GAP8 [0x00] to [0x00]
CPX: ESP32: I (434) DISCOVERY: Hostname is aideck-E0E4B4
```
Maybe this is some hint to what is going wrong?

**Reply by krichardsson** [reply, marked as answer]
The problems you get when flashing the GAP8 is most likely related to the GAP8 bootloader. It has turned out that also decks with rev D needs to be f[lashed with a new bootloader as described in the tutorial](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/#gap8-bootloader-for-the-aideck-1-0-or-aideck-1-1-rev-a-c-). I will update the tutorial.

**Comment by Kaijun101** [comment]
Hi, I am also having this problem but I do not have the Jtag programmer, therefore, I would like to check if there is alternative to resolve this?
